src/iso_alloc_sanity.c

```diff
@@ -312,6 +312,57 @@ INTERNAL_HIDDEN void *_iso_alloc_memcpy(void *restrict dest, const void *restric
     return __iso_memcpy(dest, src, n);
 }
 
+INTERNAL_HIDDEN INLINE void *__iso_memmove(void *dest, const void *src, size_t n) {
+#if MEMCPY_SANITY
+    char *p_dest = (char *) dest;
+    char const *p_src = (char const *) src;
+    size_t i;
+
+    if(p_dest > p_src && (p_dest - p_src) < n) {
+        for(i = n - 1; i >= 0; i--) {
+            p_dest[i] = p_src[i];
+        }
+    } else if(p_src > p_dest && (p_src - p_dest) < n) {
+        for(i = 0; i < n; i++) {
+            p_dest[i] = p_src[i];
+        }
+    } else {
+        dest = __iso_memcpy(dest, src, n);
```

Comment:
This is the most common case and should probably come first in the function.

Reply:
debatable ... usually memmove usage is mostly because of the upper cases :)

Reply:
glibc and musl libc implementations have differing implementations:
https://elixir.bootlin.com/glibc/latest/source/string/memmove.c
http://git.musl-libc.org/cgit/musl/tree/src/string/memmove.c

Order is less important to me than making sure we have the semantics correct.

Reply:
I took inspiration from the openbsd's implementation for the current version.